tests/sensors/test_time_sensor.py

```diff
@@ -73,3 +73,16 @@ def test_target_time_aware(self):
             op = TimeSensorAsync(task_id="test", target_time=aware_time)
             assert hasattr(op.target_datetime.tzinfo, "offset")
             assert op.target_datetime.tzinfo.offset == 0
+
+    def test_target_time_naive_dag_timezone(self):
+        """
+        Tests that naive target_time gets converted correctly using the DAG's timezone.
+        """
+        with DAG(
+            "test_target_time_naive_dag_timezone",
+            start_date=pendulum.datetime(2020, 1, 1, 0, 0, tz=DEFAULT_TIMEZONE),
+        ):
+            op = TimeSensorAsync(task_id="test", target_time=pendulum.time(8, 0))
+            assert op.target_datetime.time() == pendulum.time(0, 0)
+            assert hasattr(op.target_datetime.tzinfo, "offset")
+            assert op.target_datetime.tzinfo.offset == 0
```

Comment:
I wonder if we should use a non-zero-offset timezone to make this more obvious.

Reply:
I changed it from 9 -> 1. I'm not sure if there's a :30 timezone that is not affected by DST like Singapore isn't.